source/getting-started/control-with-amdc/encoder-fb/index.md

```diff
@@ -1,13 +1,97 @@
 # Encoder Feedback
 
+## Background
+
+Encoders provide rotor position feedback to the control system in a motor drive. This document describes a method to convert the raw readings of an encoder into meaningful position information which can be used by the contrl algorithm. Consequently, few techniques to derive the rotor speed from the measured position are also proposed. For more information on how an encoder works and how they may be interfaced with the AMDC please refer to this [document](https://docs.amdc.dev/hardware/subsystems/encoder.html#).
+
 ## Calibration
 
-- Converting from raw counts to "theta"
-- Direction: +/- depending on phase connections
-- Sync machines: dq offset
+Incremental encoders which are typically used with AMDC have a fixed number of counts per revolution (for example, 1024) and is denoted by `CPR`. The user needs needs to write some code to obtain the encoder count at a given instance and convert the count into usable angular information which can be used within the control code.
+
+### Obtaining encoder count and translating it into rotor position
+
+<img src="./resources/EncoderCodeBlockDiargam.svg" width="100%" align="center"/>
+
+As a first step, the user may use the AMDC `drv/encoder` driver module `encoder_get_position()` to get the count of the encoder reading. The`drv/encoder` driver module also has a function called `encoder_get_steps()` which gives the incremental change in the encoder position. Whereas, `encoder_get_position()` gives the actual position of the shaft and this can be converted the rotor position in radians. Next, the user needs to verify if the encoder count is increasing or decreasing with counter-clock wise rotation of shaft. This may be done by manually rotating the shaft and observing the trend of the reported position with respect to the direction of rotation. This document follows the convention of a positive rotor angle in the counter clockwise (CCW) direction of shaft rotation. Using this information, along with the offset and total encoder counts per revolution, the obtained count can be translated into angular position using a simple linear equation. The user must ensure that angle is within the bounds of 0 and 2$\pi$ by appropriately wrapping the variable using the `mod` function.
```

Comment:
I think you mean for this paragraph to be describing your figure? Please tell us that and place the paragraph *above* your figure. Start with something like "The recommended approach to reading the shaft position from the encoder is illustrated in the figure below."

Please break this up into multiple paragraphs, with each paragraph corresponding to one of the blocks in the figure:
- First paragraph is discussing obtaining the encoder reading
- Start a new paragraph at "Next, the user needs..." (to discuss direction of rotation)
- Final paragraph is discussing wrapping the encoder value with the `mod` function.

Some improvements in your figure labeling and the text of these paragraphs may be needed to help this make sense. For example, you are referring to the "offset" here, but aren't telling us what that means. Need to explain.